WalletWasabi.Fluent/Controls/TagsBox.xaml.cs

```diff
@@ -0,0 +1,354 @@
+	{
+		public static readonly StyledProperty<bool> RestrictInputToSuggestionsProperty =
+			AvaloniaProperty.Register<TagsBox, bool>(nameof(RestrictInputToSuggestions));
+
+		public static readonly StyledProperty<int> ItemCountLimitProperty =
+			AvaloniaProperty.Register<TagsBox, int>(nameof(ItemCountLimit));
+
+		public static readonly StyledProperty<object> SelectedTagProperty =
+			AvaloniaProperty.Register<TagsBox, object>(nameof(SelectedTag), defaultBindingMode: BindingMode.TwoWay);
+
+		public static readonly DirectProperty<TagsBox, IEnumerable?> SuggestionsProperty =
+			AvaloniaProperty.RegisterDirect<TagsBox, IEnumerable?>(
+				nameof(Suggestions),
+				o => o.Suggestions,
+				(o, v) => o.Suggestions = v);
+
+		public new static readonly DirectProperty<TagsBox, IEnumerable?> ItemsProperty =
+			ItemsControl.ItemsProperty.AddOwnerWithDataValidation<TagsBox>(
+				o => o.Items,
+				(o, v) => o.Items = v,
+				defaultBindingMode: BindingMode.TwoWay,
+				enableDataValidation: true);
+
+		private readonly CompositeDisposable CompositeDisposable = new CompositeDisposable();
+
+		private bool _backspaceEmptyField1;
+		private bool _backspaceEmptyField2;
+		private bool _isFocused;
+		private bool _isInputEnabled = true;
+		private IEnumerable? _items;
+		private IEnumerable? _suggestions;
+
+		private AutoCompleteBox? _autoCompleteBox;
+
+		public bool RestrictInputToSuggestions
+		{
+			get => GetValue(RestrictInputToSuggestionsProperty);
+			set => SetValue(RestrictInputToSuggestionsProperty, value);
+		}
+
+		public object SelectedTag
+		{
+			get => GetValue(SelectedTagProperty);
+			set => SetValue(SelectedTagProperty, value);
+		}
+
+		public int ItemCountLimit
+		{
+			get => GetValue(ItemCountLimitProperty);
+			set => SetValue(ItemCountLimitProperty, value);
+		}
+
+		public new IEnumerable? Items
+		{
+			get => _items;
+			set => SetAndRaise(ItemsProperty, ref _items, value);
+		}
+
+		public IEnumerable? Suggestions
+		{
+			get => _suggestions;
+			set => SetAndRaise(SuggestionsProperty, ref _suggestions, value);
+		}
+
+		protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+		{
+			base.OnApplyTemplate(e);
+
+			Presenter.ApplyTemplate();
+
+			_autoCompleteBox = (Presenter.Panel as ConcatenatingWrapPanel)?.ConcatenatedChildren
+				.OfType<AutoCompleteBox>().FirstOrDefault();
+
+			if (_autoCompleteBox is null)
+			{
+				return;
+			}
+
+			_autoCompleteBox.KeyUp += OnKeyUp;
+			_autoCompleteBox.TextChanged += OnTextChanged;
+			_autoCompleteBox.DropDownClosed += OnDropDownClosed;
+			_autoCompleteBox.GotFocus += OnOnAutoCompleteBoxGotFocus;
+			_autoCompleteBox.LostFocus += OnAutoCompleteBoxLostFocus;
+
+			_autoCompleteBox
+				.AddDisposableHandler(TextInputEvent, OnTextInput, RoutingStrategies.Tunnel)
+				.DisposeWith(CompositeDisposable);
+		}
+
+		protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+		{
+			if (_autoCompleteBox is { })
```

Comment:
There is a scenario...

1) template applied..
2) detached from visual tree..
3) attached to visual tree.

that would break this control.